src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -148,9 +150,26 @@ public function denormalize(mixed $data, string $class, ?string $format = null,
                 throw new NotNormalizableValueException('Update is not allowed for this operation.');
             }
 
-            $context[self::OBJECT_TO_POPULATE] = $this->iriConverter->getResourceFromIri($data['@id'], $context + ['fetch_data' => true]);
+            try {
+                $context[self::OBJECT_TO_POPULATE] = $this->iriConverter->getResourceFromIri($data['@id'], $context + ['fetch_data' => true]);
+            } catch (ItemNotFoundException $e) {
+                $operation = $context['operation'] ?? null;
+                if (!($operation instanceof Put && ($operation->getExtraProperties()['standard_put'] ?? false))) {
+                    throw $e;
+                }
+            }
         }
 
         return parent::denormalize($data, $class, $format, $context);
     }
+
+    protected function getAllowedAttributes(string|object $classOrObject, array $context, bool $attributesAsString = false): array|bool
+    {
+        $allowedAttributes = parent::getAllowedAttributes($classOrObject, $context, $attributesAsString);
+        if (\is_array($allowedAttributes) && ($context['api_denormalize'] ?? false)) {
+            $allowedAttributes = array_merge($allowedAttributes, ['@id', '@type', '@context']);
```

Comment:
Maybe should we allow all special keys (`@set`, `@container`, `@version`, `@nest` etc)?

Reply:
done I added all the keywords